static/js/UserHistoryPanel.jsx

```diff
@@ -145,62 +113,65 @@ UserHistoryPanel.propTypes = {
   menuOpen:            PropTypes.string.isRequired,
 };
 
-const LibraryUserHistoryPanelWrapper = ({menuOpen, toggleLanguage, openDisplaySettings, openNav, compare, toggleSignUpModal}) => {
-  return (
-    <UserHistoryPanel menuOpen={menuOpen}
-                      toggleLanguage={toggleLanguage}
-                      openDisplaySettings={openDisplaySettings}
-                      openNav={openNav}
-                      compare={compare}
-                      toggleSignUpModal={toggleSignUpModal}
-                      dataSource={'library'} />
-  )
-};
 
-const SheetsUserHistoryPanelWrapper = ({menuOpen, toggleLanguage, openDisplaySettings, openNav, compare, toggleSignUpModal}) => {
-  return (
-    <UserHistoryPanel menuOpen={menuOpen}
-                      toggleLanguage={toggleLanguage}
-                      openDisplaySettings={openDisplaySettings}
-                      openNav={openNav}
-                      compare={compare}
-                      toggleSignUpModal={toggleSignUpModal}
-                      dataSource={'sheets'} />
-  )
+const dedupeItems = (items) => {
+  /*
+  Deduplicates consecutive items with the same book or sheet_id.  
+  Essentially, we don't want two history items in a row that are of the same book (or if we're in voices, the same sheet).
+  :param items: list of UserHistory objects to deduplicate
+  :return: list of deduplicated items
+  */
+  const deduped = [];
+  const key = Sefaria.activeModule === Sefaria.VOICES_MODULE ? 'sheet_id' : 'book';  
+  let prev;
+  for (const item of items) {
+    if (item[key] !== prev) {
+      deduped.push(item);
+      prev = item[key];
+    }
+  }
+  return deduped;
 };
 
-const UserHistoryList = ({store, scrollableRef, menuOpen, toggleSignUpModal, dataSource}) => {
-  const [items, setItems] = useState(store.loaded ? store.items : null);
 
-  // Store changes when switching tabs, reset items
-  useEffect(() => {
-    setItems(store.loaded ? store.items : null);
-  }, [store]);
+const UserHistoryList = ({store, scrollableRef, menuOpen, toggleSignUpModal}) => {
```

Comment:
```suggestion
const userHistoryList = ({store, scrollableRef, menuOpen, toggleSignUpModal}) => {
```
also, all out hooks should be in Hooks.jsx